piff/stats.py

```diff
@@ -0,0 +1,264 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: stats
+
+"""
+
+from __future__ import print_function
+import numpy as np
+
+from .psf import PSF
+
+def process_stats(config, logger):
+    """Parse the stats field of the config dict.
+
+    :param config:      The configuration dict.
+    :param logger:      A logger object for logging debug info.
+
+    :returns: an stats instance
+    """
+    import piff
+
+    if 'stats' not in config:
+        raise ValueError("config dict has no stats field")
+    config_stats = config['stats']
+
+    if 'type' not in config_stats:
+        raise ValueError("config['stats'] has no type field")
+
+    # Get the class to use for the stats
+    # Not sure if this is what we'll always want, but it would be simple if we can make it work.
+    stats_class = eval('piff.' + config_stats.pop('type'))
+
+    # Read any other kwargs in the stats field
+    kwargs = stats_class.parseKwargs(config_stats)
+
+    # Build stats object
+    stats = stats_class(**kwargs)
+
+    return stats
+
+class Statistics(PSF):
```

Comment:
I don't think Statistics should be derived from PSF.  In object oriented programming public inheritance (the only kind in python) implies an "is a" relationship.  We don't have that here.  Statistics *is not a* PSF.  

I think Statistics should just be derived from the normal `object` class.  Then I would imagine that most Statistics classes would take a `PSF` instance as a construction argument.